wgsl/index.bs

```diff
@@ -4581,63 +4564,57 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in variable.
-
-    //   OpDecorate %my_pos BuiltIn Position
-    //   %float = OpTypeFloat 32
-    //   %v4float = OpTypeVector %float 4
-    //   %ptr = OpTypePointer Output %v4float
-    //   %my_pos = OpVariable %ptr Output
-
-  </xmp>
-</div>
-
-<div class='example wgsl global-scope' heading="Example built-in variable: vertex_index">
-  <xmp>
-    [[builtin(vertex_index)]] var<in> my_index : u32;
-
-    //   OpDecorate %my_index BuiltIn VertexIndex
-    //   %uint = OpTypeInt 32 0
-    //   %ptr = OpTypePointer Input %uint
-    //   %my_index = OpVariable %ptr Input
-
-  </xmp>
-</div>
-
-<div class='example wgsl global-scope' heading="Declaring other built-in variables">
+<div class='example wgsl global-scope' heading="Declaring built-in variables">
   <xmp>
-    [[builtin(instance_index)]] var<in> my_inst_index : u32;
-    //    OpDecorate %gl_InstanceId BuiltIn InstanceIndex
-
-    [[builtin(front_facing)]] var<in> is_front : u32;
-    //     OpDecorate %gl_FrontFacing BuiltIn FrontFacing
-
-    [[builtin(frag_coord)]] var<in> coord : vec4<f32>;
-    //     OpDecorate %gl_FragCoord BuiltIn FragCoord
-
-    [[builtin(frag_depth)]] var<out> depth : f32;
-    //     OpDecorate %gl_FragDepth BuiltIn FragDepth
-
-    [[builtin(local_invocation_id)]] var<in> local_id : vec3<u32>;
-    //     OpDecorate %gl_LocalInvocationID BuiltIn LocalInvocationId
-
-    [[builtin(local_invocation_index)]] var<in> local_index : u32;
-    //     OpDecorate %gl_LocalInvocationIndex BuiltIn LocalInvocationIndex
-
-    [[builtin(global_invocation_id)]] var<in> global_id : vec3<u32>;
-    //      OpDecorate %gl_GlobalInvocationID BuiltIn GlobalInvocationId
+    struct VertexOutput {
+      [[builtin(position)]] pos: vec4<f32>;
+      //   OpDecorate %my_pos BuiltIn Position
+      //   %float = OpTypeFloat 32
+      //   %v4float = OpTypeVector %float 4
+      //   %ptr = OpTypePointer Output %v4float
+      //   %my_pos = OpVariable %ptr Output
+    };
 
-    [[builtin(sample_index)]] var<in> my_sample_index : u32;
-    //      OpDecorate %gl_SampleId BuiltIn SampleId
+    [[stage(vertex)]]
+    fn main(
+      [[builtin(vertex_index)]] my_index: u32,
+      //   OpDecorate %my_index BuiltIn VertexIndex
+      //   %uint = OpTypeInt 32 0
+      //   %ptr = OpTypePointer Input %uint
+      //   %my_index = OpVariable %ptr Input
+      [[builtin(instance_index)]] my_inst_index : u32,
+      //    OpDecorate %gl_InstanceId BuiltIn InstanceIndex
+    ) -> VertexOutput;
+
+    struct FragmentOutput {
+      [[builtin(frag_depth)]] depth: f32;
+      //     OpDecorate %gl_FragDepth BuiltIn FragDepth
+      [[builtin(sample_mask_out)]] mask_out : u32;
+      //      OpDecorate %gl_SampleMask BuiltIn SampleMask ; an output variable
+    };
 
-    [[builtin(sample_mask_in)]] var<in> mask_in : u32;
-    //      OpDecorate %gl_SampleMaskIn BuiltIn SampleMask ; an input variable
-    //      OpDecorate %gl_SampleMaskIn Flat
+    [[stage(fragment)]]
+    fn main(
```

Comment:
These functions should all have unique names, so probably `vert_main`, `frag_main` and `comp_main` so that we can compile the example.

Reply:
would be great to compile it on CI ;)

Reply:
@ben-clayton has a script to run them through Tint, but it would be tricky in most cases as the implementation lags the spec, so new spec features would always show up as broken.